api/handlers.go

```diff
@@ -698,6 +698,7 @@ func postContainersCreate(c *context, w http.ResponseWriter, r *http.Request) {
 		httpError(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	containerConfig.NetworkingConfig = cluster.CleanupNetworkingConfig(containerConfig.NetworkingConfig, c.cluster.EngineNames())
```

Comment:
I was a bit on the fence if we should move this into `cluster.CreateContainer()`; that way `CreateContainer` would responsible for dealing with this (and we wouldn't have to modify the Interface to add `EngineNames()`).

I went looking for uses of `CreateContainer()` and the other location it's used is in [`rescheduleContainers`](https://github.com/docker/swarm/blob/0a268e4925fb65c51a13f1482082ca5b4028aeb0/cluster/watchdog.go#L144-L145), which copies the configuration of existing containers, so in that case the nodename would be stripped _twice_

We could consider making it a local utility and move it here, given that it's fixing something specific to this endpoint, but happy to get your thought on that

Reply:
I think it makes sense to just strip the node names when the container configs first come into the API; as you point out, we should definitely not be stripping node names in `rescheduleContainers`. I'll move `CleanupNetworkingConfig` (aka `StripNodeNames`) into this file.